homeassistant/components/http/__init__.py

```diff
@@ -340,21 +344,16 @@ def register_static_path(self, url_path, path, cache_headers=True):
         # To work around this we now prevent the router from getting frozen
         self.app._router.freeze = lambda: None
 
-        self._handler = self.app.make_handler(loop=self.hass.loop)
-
+        self.runner = web.AppRunner(self.app)
+        await self.runner.setup()
+        site = web.TCPSite(self.runner, self.server_host, self.server_port,
+                           ssl_context=context)
         try:
-            self.server = await self.hass.loop.create_server(
-                self._handler, self.server_host, self.server_port, ssl=context)
+            await site.start()
         except OSError as error:
             _LOGGER.error("Failed to create HTTP server at port %d: %s",
                           self.server_port, error)
 
     async def stop(self):
-        """Stop the WSGI server."""
-        if self.server:
-            self.server.close()
-            await self.server.wait_closed()
-        await self.app.shutdown()
-        if self._handler:
-            await self._handler.shutdown(10)
-        await self.app.cleanup()
+        """Stop the aiohttp server."""
+        await self.runner.cleanup()
```

Comment:
https://github.com/home-assistant/hassio/blob/dev/hassio/api/__init__.py#L254-L256

Reply:
Docs don't mention closing the site

https://aiohttp.readthedocs.io/en/stable/web_advanced.html#application-runners

Reply:
Yes, but I found also no code they call that function by aiohttp. It is maintained by TCPSite. I'm not sure what will be correct but I known that the code should be called to cleanup that correctly.

Reply:
Added site.stop